### PR TITLE
Feanil/vpc tool update

### PR DESCRIPTION
**Pull request by feanil**

### CODE REVIEW

File: playbooks/library/util_map [hunk 1]
```diff
@@ -103,10 +103,37 @@ def zip_to_dict(module, input, key_key, value_key):
...
+    for item in input:
+        results.append(item[key])
+
+    module.exit_json(function_output = results)
+
 def main():
 
     arg_spec = dict(
-        function=dict(required=True,choices=['zip_to_dict','flatten']),
+        function=dict(required=True, type='str'),
```
Comment (e0d) [inline playbooks/library/util_map:136]: why this change?

File: playbooks/roles/edx_vpc/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  ec2_rt:
+    profile: "{{ vpc_aws_profile }}"
+    vpc_id: "{{ created_vpc.vpc_id }}"
+    region: "{{ vpc_aws_region }}"
+    state: "present"
+    name: "{{ vpc_name }}-private"
+    routes: "{{ vpc_private_route_table }}"
+  register: created_private_rt
+
+- name: create db network acl
```
Comment (e0d) [inline playbooks/roles/edx_vpc/tasks/main.yml:149]: If we're going to have a db per service, this probably belongs in the specific service.

File: playbooks/roles/edx_vpc/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+# github:     https://github.com/edx/configuration
+# wiki:       https://github.com/edx/configuration/wiki
+# code style: https://github.com/edx/configuration/wiki/Ansible-Coding-Conventions
+# license:    https://github.com/edx/configuration/blob/master/LICENSE.TXT
+#
+#
+#
+# Tasks for role edx_vpc
+# 
+# Overview:
```
Comment (e0d) [inline playbooks/roles/edx_vpc/tasks/main.yml:14]: Need some doc, should outline out opinionated assumptions that are baked in here.

File: playbooks/roles/edx_vpc/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (maxrothman) [inline playbooks/roles/edx_vpc/handlers/main.yml]: I thought we don't use handlers anymore?

File: playbooks/roles/edx_vpc/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (e0d) [inline playbooks/roles/edx_vpc/meta/main.yml]: If not needed, remove.

File: playbooks/roles/edx_vpc/handlers/main.yml (hunk 4 above)
Comment (e0d) [inline playbooks/roles/edx_vpc/handlers/main.yml]: Remove

File: playbooks/library/util_map (hunk 1 above)
Comment (feanil) [inline playbooks/library/util_map:136]: Otherwise we have to add each new type of function to this list, and to the if statement below which seemed a bit repetitive.

File: playbooks/edx-east/edx_vpc.yml [hunk 6]
```diff
@@ -1,120 +1,10 @@
...
-      local_action:
-        module: "ec2_lookup"
-        region: "{{ aws_region }}"
-        tags:
-          Name: "{{ vpc_name }}-nat-instance"
-      register: potential_existing_nat
-
-    - debug: msg="{{ potential_existing_nat }}"
-
-    - name: create nat instance
```
Comment (maxrothman) [inline playbooks/edx-east/edx_vpc.yml:83]: Nitpick on things in this task: inconsistent quoting

File: playbooks/edx-east/edx_vpc.yml [hunk 7]
```diff
@@ -1,120 +1,10 @@
...
-        state: 'present'
-        wait: "yes"
-        source_dest_check: false
-        region: "{{ aws_region }}"
-        profile: "{{ profile }}"
-        group_id: "{{ created_nat_security_group.group_id }}"
-        key_name: "{{ vpc_nat_keypair }}"
-        vpc_subnet_id: "{{ created_public_subnets.results[0].subnet_id }}"
-        instance_type: "{{ vpc_nat_instance_type }}"
-        instance_tags:
```
Comment (maxrothman) [inline playbooks/edx-east/edx_vpc.yml:95]: Remove

File: playbooks/roles/edx_vpc/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    key_name: "{{ vpc_keypair }}"
+    vpc_subnet_id: "{{ created_public_subnets.results[0].subnet_id }}"
+    instance_type: "{{ vpc_nat_instance_type }}"
+    instance_tags:
+      Name: "{{ vpc_name }}-nat-instance"
+    image: "{{ vpc_nat_ami_id }}"
+  register: new_nat_instance
+  when: nat_instance.instances|length == 0
+
+# We need to do this instead of registering the output of the above
```
Comment (maxrothman) [inline playbooks/roles/edx_vpc/tasks/main.yml:119]: If so, then why do you use `new_nat_instance` below?

Comment (feanil) [inline playbooks/roles/edx_vpc/tasks/main.yml:119]: Because the EIP module is not actually idempotent right now so it should only be run if a new instance is created.

**Comment by e0d** [comment]
:+1:
